ui/wallet_page.go

```diff
@@ -46,329 +55,467 @@ func (win *Window) WalletPage(common pageCommon) layout.Widget {
+							accountsName := accounts[x].Name
+							totalBalance := accounts[x].TotalBalance
+							spendable := dcrutil.Amount(accounts[x].SpendableBalance).String()
+
+							click := pg.toAcctDetails[x]
+							pointer.Rect(image.Rectangle{Max: gtx.Constraints.Max}).Add(gtx.Ops)
+							click.Add(gtx.Ops)
+							pg.goToAcctDetails(gtx, common, &accounts[x], wIndex, click)
+							return pg.walletAccountsLayout(gtx, accountsName, totalBalance, spendable, common)
+						})
+					}),
+					layout.Rigid(func(gtx C) D {
+						pg.line.Width = gtx.Constraints.Max.X
+						pg.line.Color = common.theme.Color.Background
+						m := values.MarginPadding10
+						return layout.Inset{Top: m, Bottom: m}.Layout(gtx, func(gtx C) D {
+							return pg.line.Layout(gtx)
+						})
+					}),
+					layout.Rigid(func(gtx C) D {
+						return layout.Flex{}.Layout(gtx,
+							layout.Rigid(func(gtx C) D {
+								return layout.Inset{
+									Bottom: values.MarginPadding5,
+									Right:  values.MarginPadding10,
+								}.Layout(gtx, func(gtx C) D {
+									return pg.addAcct[i].Layout(gtx)
+								})
+							}),
+							layout.Rigid(func(gtx C) D {
+								txt := pg.theme.H6("Add new account")
+								txt.Color = common.theme.Color.Gray
+								return txt.Layout(gtx)
+							}),
+						)
+					}),
+				)
+			})
+		}
+
+		return layout.Inset{Bottom: values.MarginPadding15}.Layout(gtx, func(gtx C) D {
+			if len(seed) > 0 {
+				return pg.walletCollapsible[i].Layout(gtx, collapsibleHeader, collapsibleBody, collapsibleFooter)
+			}
+			return pg.walletCollapsible[i].Layout(gtx, collapsibleHeader, collapsibleBody, nil)
+		})
+	})
+}
+
+func (pg *walletPage) watchOnlyWalletSection(gtx layout.Context, common pageCommon) layout.Dimensions {
+	return layout.Inset{Top: values.MarginPadding5}.Layout(gtx, func(gtx C) D {
+		return pg.sectionLayout(gtx, func(gtx C) D {
+			return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+				layout.Rigid(func(gtx C) D {
+					txt := pg.theme.Body1("Watch-only Wallets")
+					txt.Color = pg.theme.Color.Gray
+					return txt.Layout(gtx)
+				}),
+				layout.Rigid(func(gtx C) D {
+					pg.line.Width = gtx.Constraints.Max.X
+					pg.line.Color = common.theme.Color.Hint
+					m := values.MarginPadding10
+					inset := layout.Inset{
+						Top:    m,
+						Bottom: m,
+					}
+					return inset.Layout(gtx, func(gtx C) D {
+						return pg.line.Layout(gtx)
+					})
+				}),
 				layout.Rigid(func(gtx C) D {
-					return common.theme.H5(pg.current.Name).Layout(gtx)
+					return pg.theme.H6("coming soon").Layout(gtx)
 				}),
 			)
-		},
-		func(gtx C) D {
-			return common.theme.H6("Total Balance: " + pg.current.Balance).Layout(gtx)
-		},
-		func(gtx C) D {
-			return layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
+		})
+	})
+}
+
+func (pg *walletPage) tableLayout(gtx layout.Context, leftLabel, rightLabel decredmaterial.Label, isIcon bool, seed int) layout.Dimensions {
```

Comment:
This method has a generic name. The code however suggests that it executes specific layouts. Perhaps the code could be simpler?